### ISSUE TITLE
[Question]: after login the redirect url not work, always redirect to project home page

### ISSUE THREAD

**Issue by xychen362** [damienbod/angular-auth-oidc-client#1757]
**What Version of the library are you using?**
15.0.4

**Question**
hello, i have set the AuthModule.forRoot's config, it defined the redirectUrl to 'http://localhost:4200/callback'. The callback is one of my component. and i added a route guard for callback component so i can add some logic when catching the redirect action.

1) however after i login, the route just **temporary** redirect to callback then it will jump to project home page. 
2) another strange thing is in when redirect to the callback, **route guard it can't get the token.**  i add some log and found it will go to guard firstly and then the EventTypes.NewAuthenticationResult get the token.
 is there something i made wrong? Thanks

**Comment by FabianGosebrink** [comment]
The callback should not have a guard, because it is needed to have a component to let the app set up everything with the method `checkAuth()`. _After_ this is done, you can get the token and everything out of the lib.

**Comment by xychen362** [comment]
> The callback should not have a guard, because it is needed to have a component to let the app set up everything with the method `checkAuth()`. _After_ this is done, you can get the token and everything out of the lib.

**Comment by xychen362** [comment]
thanks for the reply, does this mean we can consider the callback component or other component we used as the redirectUrl is a middle component, shouldn't handle too much logic in it?

**Comment by FabianGosebrink** [comment]
Exactly. This is just one public starting point to give the app and lib a chance to start up.

**Comment by xychen362** [comment]
got it. :)   then what about another issue, after the feedback, i remove the guard for callback, but it still continue direct to the homepage component instead of stay to the callback component. the callback component currently have nothing but just a initial empty component

**Comment by jerousseau** [comment]
Before redirecting to login page the guard save the route you try to access and will redirect to this route after callback (auto-login.service.ts).
In case of use in a CanLoad guard (Lazy load component / deprecated in angular 15.1 ) the url stored is an empty string so the autologin won't redirect and the user stay on the callback component, I thinks the auto-login.service may check if the savedRouteForRedirect is null and redirect in case of empty string (@FabianGosebrink )

**Comment by xychen362** [comment]
hello, thanks for the comments and sorry for the late reply, i had a vacation last week. 

based on the comment, i build another component called neededRedirectAfterlogin, and then i type address localhost:4200/neededRedirectAfterlogin, after it jump to login page, i request to login, and the expected behavior is after login the app would redirect to neededRedirectAfterlogin, but instead, it still went to dashboard which is the app's default route. Is there something i miss? what should i do if i want to control where to redirect after login?  Looking forward to your reply. (@jerousseau @FabianGosebrink)

**Comment by FabianGosebrink** [comment]
You have to configure this in your IDP as allowed route to redirect and pass this route in the config then.

**Comment by xychen362** [comment]
hello now it can redirect correctly. but there are still some confusion. here is my AuthModule.forRoot:
`AuthModule.forRoot({
      config: {
        authority: environment.SAuthConfig.authority,
        redirectUrl: 'http://localhost:4200/callback',
        postLogoutRedirectUri: 'http://localhost:4200/callback',
        clientId: environment.SAuthConfig.clientId,
        scope: 'openid',
        responseType: 'code',
        silentRenew: true,
        useRefreshToken: true,
        logLevel: LogLevel.Debug,
      },
    }),`
and this is the app-routing config:
`const routes: Routes = [
    {
        path: 'login',
        component: LoginComponent,
    },
    {
        path: 'dashboard',
        component: DashboardComponent,
        canActivate: [AutoLoginPartialRoutesGuard],
    },
    {
        path: 'callback',
        component: CallbackComponent,
    },
    {
        path: 'neededRedirectAfterlogin',
        component: NeededRedirectAfterloginComponent,
        canActivate: [AutoLoginPartialRoutesGuard]
    },
    {
        path: '',
        pathMatch: 'full',
        redirectTo: 'dashboard',
    },
    {
        path: 'unsubscribe',
        component: UnsubscribeComponent,
        canActivate: [AutoLoginPartialRoutesGuard]
    },
];`

this is the checkAuth in appComponent onInit()
`this.oidcSecurityService.checkAuth().subscribe((result) => {
        if(result.isAuthenticated) {
            this.router.navigate(['/neededRedirectAfterlogin']);
        } else {
            this.router.navigate(['/login']);
        }
    })`
it can redirect to neededRedirectAfterlogin,  but from the callback it flash to dashboard, and then go to neededRedirectAfterlogin, is there a way let it directly goes to neededRedirectAfterlogin? or i miss some configuration?

**Comment by python-programmer** [comment]
add `postLoginRoute: '/neededRedirectAfterlogin,` to your config
